task8/src/main/java/PostCollection.java

```diff
@@ -0,0 +1,157 @@
+        List<Post> filteredPosts = new ArrayList<>();
+        for (Map.Entry pair : filterConfig.entrySet()) {
+            if (pair.getKey().equals(KEY_AUTHOR)) {
+                if (pair.getValue() instanceof String) {
+                    posts.stream()
+                            .filter(post -> post.getAuthor().toLowerCase().contains(pair.getValue().toString().toLowerCase()))
+                            .forEach(filteredPosts::add);
+                }
+            } else if (pair.getKey().equals(KEY_CREATION_DATE)) {
+                if (pair.getValue() instanceof Date) {
+                    posts.stream()
+                            .filter(post -> post.getCreationDate().equals(pair.getValue()))
+                            .forEach(filteredPosts::add);
+                }
+            } else if (pair.getKey().equals(KEY_VALIDATE_UNTIL)) {
+                if (pair.getValue() instanceof Date) {
+                    posts.stream()
+                            .filter(post -> post.getValidateUntil().equals(pair.getValue()))
+                            .forEach(filteredPosts::add);
+                }
+            } else if (pair.getKey().equals(KEY_HASHTAG)) {
+                if (pair.getValue() instanceof String) {
+                    posts.stream()
+                            .filter(post -> Arrays.asList(post.getHashTags()).contains(pair.getValue()))
+                            .forEach(filteredPosts::add);
+                }
+            } else if (pair.getKey().equals(KEY_RATING)) {
+                if (pair.getValue() instanceof Integer) {
+                    posts.stream()
+                            .filter(post -> post.getRating() == (int) pair.getValue())
+                            .forEach(filteredPosts::add);
+                }
+            } else if (pair.getKey().equals(KEY_DISCOUNT)) {
+                if (pair.getValue() instanceof Integer) {
+                    posts.stream()
+                            .filter(post -> post.getDiscount() == (int) pair.getValue())
+                            .forEach(filteredPosts::add);
+                }
+            }
+        }
+        if (filterConfig.size() == 0) {
+            filteredPosts = new ArrayList(posts);
+        }
+        if (top > filteredPosts.size()) {
+            top = filteredPosts.size();
+        }
+        if (skip >= filteredPosts.size()) {
+            skip = 0;
+        }
+        if (skip + top > filteredPosts.size()) {
+            return filteredPosts.subList(skip, filteredPosts.size());
+        } else {
+            return filteredPosts.subList(skip, skip + top);
+        }
+    }
+
+
+    public Post getPost(int id) {
+        for (Post post : posts) {
+            if (post.getId() == id) {
+                return post;
+            }
+        }
+        return null;
+    }
+
+    public boolean validatePost(Post post) {
+        for (Post currentPost : posts) {
```

Comment:
It's better to encapsulate `isValid()` as a method in the very `Post` class.
Also, the check for "post already present" in `validate()` a bit breaks the semantics/readability. To be "already present" does not mean to be "invalid". These are two different "qualities" of an object